From c361db0053a02e6c38391aa62d66ac3a68069f97 Mon Sep 17 00:00:00 2001
From: Vaibhav Bedia <vaibhav.bedia@ti.com>
Date: Wed, 19 Jan 2011 19:46:28 +0530
Subject: [PATCH] ti8148: Add the 2nd stage bootcmd

This patch adds the 2nd bootcmd to execute a script from MMC/SD card

Signed-off-by: Vaibhav Bedia <vaibhav.bedia@ti.com>
---
 include/configs/ti8148_evm.h |   24 ++++++++++++++++++++++++
 1 files changed, 24 insertions(+), 0 deletions(-)

diff --git a/include/configs/ti8148_evm.h b/include/configs/ti8148_evm.h
index ebdc3fe..931be74 100644
--- a/include/configs/ti8148_evm.h
+++ b/include/configs/ti8148_evm.h
@@ -77,6 +77,30 @@
 # define CONFIG_NAND			1
 # define CONFIG_SPI			1
 # define CONFIG_I2C			1
+# define CONFIG_EXTRA_ENV_SETTINGS \
+	"verify=yes\0" \
+	"bootfile=uImage\0" \
+	"ramdisk_file=ramdisk.gz\0" \
+	"loadaddr=0x81000000\0" \
+	"script_addr=0x80900000\0" \
+	"loadbootscript=fatload mmc 1 ${script_addr} boot.scr\0" \
+	"bootscript= echo Running bootscript from MMC/SD to set the ENV...; " \
+		"source ${script_addr}\0" \
+
+# define CONFIG_BOOTCOMMAND \
+	"if mmc init; then " \
+		"if run loadbootscript; then " \
+			"run bootscript; " \
+		"else " \
+			"echo In case ENV on MMC/SD is required; "\
+			"echo Please put a valid script named boot.scr on the card; " \
+			"echo Refer to the User Guide on how to generate the image; " \
+		"fi; " \
+	"else " \
+		"echo Please set bootargs and bootcmd before booting the kernel; " \
+		"echo If that has already been done please ignore this message; "\
+	"fi"
+
 
 #endif
 
-- 
1.6.2.4

